src/compile/selection/point.ts

```diff
@@ -1,11 +1,49 @@
-import {Stream} from 'vega';
+import {Signal, Stream} from 'vega';
 import {stringValue} from 'vega-util';
-import {SelectionCompiler, TUPLE, unitName} from '.';
+import {SelectionCompiler, TUPLE, isTimerSelection, unitName} from '.';
 import {SELECTION_ID} from '../../selection';
 import {vals} from '../../util';
 import {BRUSH} from './interval';
 import {TUPLE_FIELDS} from './project';
 
+export const CURR = '_curr';
+
+const animationSignals = (selectionName: string, scaleName: string): Signal[] => {
+  return [
```

Comment:
Are these selection-specific signals, or global signals (i.e., one set of signals that are used by all selections)? I _think_ it's the former based on some of the references within. In which case, I think they'll all need to be prefixed by the selection name. Otherwise, if you have multiple timer selections within the same unit spec, you're going to get a duplicate signal error.

Reply:
i think deciding what's selection-specific vs global matters for two situations:

1. multiple timer selections in the same unit spec

i don't think this should be allowed, because what does it mean to animate the a single set of marks two different ways? what's the best way to disallow this?

2. multiple timer selections across multiple unit specs in a multi-view

we ideally want the animation clock `ANIM_CLOCK` to be global, so we can share a common timer across multiple units. but the problem is that the animation clock needs to know the `MAX_RANGE_EXTENT` to know when to wrap back around to 0, but this is calculated from the scale for each unit. imo for now this is fine since we didn't consider multi-view in the scope of the paper, but we'll have to figure this out when we think about multi-view / scale resolution. the extent of the clock needs to be determined from the unioned scale or the max of the independent scale extents

Reply:
Re: (2) a global `ANIM_CLOCK`, I agree that we didn't consider multi-view in the paper, but I feel a little strange to introduce this constrain in production since, from an end user perspective, it might seem like an arbitrary limitation? Could you humor me and help me think through how complex this would be to implement? If the issue is primarily about how to resolve `MAX_RANGE_EXTENT` for wrapping around back to 0, are there more than these two cases we would need to handle:

1. No global `ANIM_CLOCK`: so each unit can have only one timer selection, and they are independent and loop independently.
2. A shared `ANIM_CLOCK`: they would wrap around at the same time; that means, you essentially would need to max over the constituent `MAX_RANGE_EXTENT` and then use that to determine the wrap around. The effect would be that one timer would "pause" while the other(s) finish, and then they wrap.

These two cases feel like they map well to VL's existing notion of `resolve` (which is used for scales/axes) and which we overload for selection predicates in terms of `union`/`intersect`. So I feel like there's conceptual machinery already in place that we might be able to piggy back off?

Reply:
logging and unit tests added

https://github.com/vega/vega-lite/pull/8921/files/25d1b1c02b4aac8c2cbc72758eb36dda7cedb2ef..c2b4b59bad9f2e6b6b3fc180c5009e56dd61fd92

Reply:
option 2 sounds like what we want eventually. i looked back at what we did for the paper and we did implement `layer` and vconcat` in the prototypes. but in our prototypes these were two separate implementations with not a lot of overlap. i also spent most of the day trying to get facet to work, and it almost works modulo the range resolution stuff, but i have no idea how to do it. we knew when we were doing the paper that conceptually there's overlap, but i suspect there isn't overlap in the implementation (at least right now) since we're using signals and note stores, and can't make use of `vlSelectionResolve` directly

all in all, i think it'd be a pretty big lift to get multi-view working in this PR. when we set out to do this PR, we thought of it as the minimal set of features to get people making basic animations that would cover the most common use case (we also aren't scoping in things like interpolation which i would prioritize over multi-view). i'm hesitant to increase the scope at this point because of 1) how hard it is for me to build velocity in this codebase due to complexity 2) the accessibility libraries being more of a personal priority